mongoengine/connection.py

```diff
@@ -7,6 +7,10 @@
 
 
 DEFAULT_CONNECTION_NAME = 'default'
+if pymongo.version_tuple[0] >= 3:
+    READ_PREFERENCE = ReadPreference.SECONDARY_PREFERRED
```

Comment:
This should be `PRIMARY` by default.

`SECONDARY_PREFERRED` as a default with no other configuration will break many applications - as it introduces an inherent race conditions for replica set systems.  Imagine adding a User to the database and then on the next request trying to read from a secondary to get that User - it might not have been replicated to the secondary node.